Fig7_PyGMT_datetime.py

```diff
@@ -0,0 +1,76 @@
+import requests
+import pandas as pd
+
+def get_star_history(repo):
+    owner = "GenericMappingTools"
+    headers = {
+        "Accept": "application/vnd.github.v3.star+json",
+        # "Authorization": "Bearer {YOUR_GITHUB_TOKEN_HERE}",
+    }
+
+    timestamps, users = [], []
+    page = 1
+
+    while True:
+        r = requests.get(
+            f"https://api.github.com/repos/{owner}/{repo}/stargazers",
+            headers=headers,
+            params={"per_page": 100, "page": page},
+        )
+        data = r.json()
+        if not data:
+            break
+        # print(data)
+
+        timestamps += [s["starred_at"] for s in data]  # full ISO 8601 timestamp
+        users += [s["user"]["login"] for s in data]
+        page += 1
+    df = pd.DataFrame({"timestamp": timestamps, "user": users})
+    df.sort_values("timestamp", inplace=True)
+
+    df.to_csv(f"star_history_github_{repo}.csv", sep=",", index=False)
+
+
+for repo in ["gmt", "pygmt", "gmt.jl", "gmtmex"]:
+    get_star_history(repo)
+
+# %%
+import datetime
+
+import pandas as pd
+import pygmt
+from pygmt.params import Box
+
+stars_step = 20
+
+fig = pygmt.Figure()
+fig.basemap(
+    projection="X12c/6c",
+    region=[datetime.date(2016, 1, 1), datetime.date(2026, 12, 31), -50, 1000],
+    frame=["x+lYear", "ya100f50+lGitHub stars"],
+)
+
+for wrapper, file, color, symbol in zip(
+    ["GMT", "PyGMT", "GMT.jl", "GMT/MEX"],
+    ["gmt", "pygmt", "gmt.jl", "gmtmex"],
+    ["238/86/52", "48/105/152", "149/88/178", "230/51/51"],
+    ["C", "A", "T", "I"],
+    strict=False,
+):
+    df = pd.read_csv(
+        f"star_history_github_{file}.csv",
+        sep=",",
+        parse_dates=["timestamp"],
+        index_col="timestamp",
+    )
+    df = df.resample("3ME").count().cumsum().rename(columns={"user": "stars"})

```

Comment:
What about 6-month interval?
```suggestion
    df = df.resample("6ME").count().cumsum().rename(columns={"user": "stars"})
```

Reply:
Support to use 6-month intervals, currently the figure looks quite busy with the many data points.